addons/portal/models/ir_qweb_fields.py

```diff
@@ -0,0 +1,14 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import models
+from odoo.http import request
+
+
+class Image(models.AbstractModel):
+    _inherit = 'ir.qweb.field.image'
+
+    def _get_src_urls(self, record, field_name, options):
+        src, src_zoom = super()._get_src_urls(record, field_name, options)
+        if isinstance(record, self.env.registry['portal.mixin']) and request and request.params.get('access_token'):
+            src += '&access_token=' + request.params.get('access_token')
```

Comment:
I find it odds to reuse the access_token from the request for the new url, shouldn't it be:
```suggestion
        if isinstance(record, self.env.registry['portal.mixin']):
            src += '&access_token=' + record.access_token
```

Reply:
I agree it's odd to use `request.params.get('access_token')`, but to me it's also odd to use`record.access_token` because at that point you don't know if the record was accessed using an access token or not.
Meaning, if you printed the report without access token, it's odd to pass the `record.access_token` to get the images.

Ìs there not a way to get the access token from the passed context to QWeb ?